File: src/kyu_6/validate_credit_card_number/c/validate_credit_card_number.c
#include <stdbool.h>

bool validate(long digits)
{
  int sum = 0, j = 0;
  for (long n = digits; n; n /= 10)
  {
    int d = n % 10;
    if (j++ % 2) d *= 2;
    if (d > 9) d -= 9;
    sum += d;
  }
  return sum % 10 == 0;
}

File: src/kyu_6/validate_credit_card_number/python/validate_credit_card_number.py
def validate(n):
    digits = [int(x) for x in str(n)]
    even = [x*2 if x*2 <= 9 else x*2 - 9 for x in digits[-2::-2]]
    odd  = [x for x in digits[-1::-2]]
    return (sum(even + odd)%10) == 0